crates/astria-cli/src/sequencer/threshold/sign.rs

```diff
@@ -0,0 +1,274 @@
+};
+
+use super::read_line_raw;
+
+#[derive(Debug, clap::Args)]
+pub(super) struct Command {
+    #[command(subcommand)]
+    command: SubCommand,
+}
+
+impl Command {
+    pub(super) async fn run(self) -> eyre::Result<()> {
+        match self.command {
+            SubCommand::Part1(part1) => part1.run(),
+            SubCommand::PrepareMessage(prepare_message) => prepare_message.run().await,
+            SubCommand::Part2(part2) => part2.run(),
+            SubCommand::Aggregate(aggregate) => aggregate.run().await,
+        }
+    }
+}
+
+#[derive(Debug, clap::Subcommand)]
+enum SubCommand {
+    /// perform part 1 of the signing protocol.
+    ///
+    /// generates participant commitments (used in `prepare-message`)
+    /// and nonces (used in part 2).
+    Part1(Part1),
+
+    /// generate a signing package given a message to be signed
+    /// and commitments from part 1.
+    ///
+    /// can be executed by a coordinator, or by every individual participant, assuming
+    /// all participants use the same commitments generated in part 1.
+    PrepareMessage(PrepareMessage),
+
+    /// perform part 2 of the signing protocol.
+    ///
+    /// generates a signature share using the nonces from part 1 and the
+    /// signing package from `prepare-message`.
+    Part2(Part2),
+
+    /// aggregate signature shares from part 2 to produce the final signature.
+    Aggregate(Aggregate),
+}
+
+#[derive(Debug, clap::Args)]
+struct Part1 {
+    /// path to a file with the secret key package from keygen ceremony
+    #[arg(long)]
+    secret_key_package_path: String,
+
+    /// path to a file to output the nonces
+    #[arg(long)]
+    nonces_path: String,
+}
+
+impl Part1 {
+    fn run(self) -> eyre::Result<()> {
+        let mut rng = thread_rng();
+
+        let Self {
+            secret_key_package_path,
+            nonces_path,
+        } = self;
+
+        let secret_package = serde_json::from_slice::<frost_ed25519::keys::KeyPackage>(
+            &std::fs::read(secret_key_package_path)
+                .wrap_err("failed to read secret key package file")?,
+        )?;
+        let (nonces, commitments) =
+            frost_ed25519::round1::commit(secret_package.signing_share(), &mut rng);
+        let commitments_with_id = CommitmentsWithIdentifier {
+            identifier: *secret_package.identifier(),
+            commitments,
+        };
+
+        println!("Writing nonces to {nonces_path}");
+        std::fs::write(nonces_path, hex::encode(nonces.serialize()?).as_bytes())?;
+
+        println!(
+            "Our commitments are: {}",
+            serde_json::to_string(&commitments_with_id)?
+        );
+        Ok(())
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct CommitmentsWithIdentifier {
+    identifier: Identifier,
+    commitments: frost_ed25519::round1::SigningCommitments,
+}
+
+#[derive(Debug, clap::Args)]
+struct PrepareMessage {
+    /// message to be signed
+    #[arg(long)]
+    message: String,
```

Comment:
Do we need something that is not a string for the message?

This message is generally going to be the transaction we are signing right? At some point it needs to become a protobuf serialized `TransactionBody` (formerly `UnsignedTransaction`)? 

In the simplest form we could have a message which is just pbjson via a file read? Then we might want to add a way to generate txs without signing and output the pbjson form of them, but not that's more better ux, we could generate that ourselves relatively simply?

Reply:
updated this to take a file that contains the message bytes. I could add an option to deserialize the message as a `TransactionBody` and display that before signing?

> This message is generally going to be the transaction we are signing right?

yes, I was thinking of updating the other CLI commands to have an option to output the `TransactionBody` into a file, which is then used as the input to this command. 

I also updated `aggregate` to have an option to output the signed message as a `Transaction` (signed) that can then be submitted to the sequencer.